cli/core/test/e2e/plugin_lifecycle/plugin_lifecycle_suite_test.go

```diff
@@ -0,0 +1,13 @@
+package plugin
+
```

Comment:
wondering: is this the folder where actual plugin lifecycle e2e test cases will be added in the future? 
The only tests in this folder so far in this PR is about exercising the framework functionality.
Have you considered keeping these tests in framework/ instead?

Reply:
Yes, the plan is to add plugin life cycle tests in the plugin package. i thought of having only framework-related and helper code in the framework package and having separate packages for the actual tests.

Reply:
I have updated the plugin generation and published the code to setup node.